include/boost/url/query_param.hpp

```diff
@@ -172,12 +172,12 @@ struct query_param_encoded_view
     {
         if(has_value)
             return {
-                pct_encoded_view(key),
-                pct_encoded_view(value),
+                decode_view(key),
+                decode_view(value),
```

Comment:
Something isn't right here, because `decode_view` will do validation and throw on invalid input. But if this object is being constructed by the container, we already know it is valid. Furthermore, the constructor for `decode_view` can throw, but is being called inside a `noexcept` function.

Reply:
Yes. We have many issues here. If we want to ensure this is only created by the container, then we need to hide the public constructors that don't validate things. If we want to the user to be able to create this, to either validate the input when constructing it or remove `noexcept` from the conversion.